thrust/testing/cuda/sort.cu

```diff
@@ -157,3 +167,95 @@ struct TestRadixSortDispatch
 // TODO(bgruber): use a single test case with a concatenated key list and a cartesion product with the comparators
 SimpleUnitTest<TestRadixSortDispatch, IntegralTypes> TestRadixSortDispatchIntegralInstance;
 SimpleUnitTest<TestRadixSortDispatch, FloatingPointTypes> TestRadixSortDispatchFPInstance;
+
+/**
+ * Copy of CUB testing utility
+ */
+template <typename UnsignedIntegralKeyT>
+struct index_to_key_value_op
+{
+  static constexpr std::size_t max_key_value =
+    static_cast<std::size_t>(::cuda::std::numeric_limits<UnsignedIntegralKeyT>::max());
+  static constexpr std::size_t lowest_key_value =
+    static_cast<std::size_t>(::cuda::std::numeric_limits<UnsignedIntegralKeyT>::lowest());
+  static constexpr std::size_t num_distinct_key_values = (max_key_value - lowest_key_value + std::size_t{1ULL});
+
+  __device__ __host__ UnsignedIntegralKeyT operator()(std::size_t index)
+  {
+    return static_cast<UnsignedIntegralKeyT>(index % num_distinct_key_values);
+  }
+};
+
+/**
+ * Copy of CUB testing utility
```

Comment:
Remark: I would really like to have a unified testing infrastructure between Thrust and CUB, and probably cudax as well.

Reply:
Agreed! We need to extract a few parts of our testing and benchmarking infrastructure into reusable components. Ideally, we'd extract data generation facility into its own header to be reused across tests and benchmarks as well. We'll get to this once other higher priority issues are addressed.